// @keystonejs/fields-wysiwyg-tinymce/dist/fields-wysiwyg-tinymce.cjs.js
'use strict';

var _join = require("path").join;

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopDefault(ex) {
  return ex && typeof ex === 'object' && 'default' in ex ? ex['default'] : ex;
}

var path = require('path');

var express = _interopDefault(require('express'));

var fields = require('@keystonejs/fields');

var _defineProperty = _interopDefault(require("@babel/runtime/helpers/defineProperty"));

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);
    if (enumerableOnly) symbols = symbols.filter(function (sym) {
      return Object.getOwnPropertyDescriptor(object, sym).enumerable;
    });
    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

class WysiwygImplementation extends fields.Text.implementation {
  constructor(path, {
    editorConfig
  }) {
    super(...arguments);
    this.editorConfig = editorConfig;
  }

  extendAdminMeta(meta) {
    return _objectSpread({}, meta, {
      editorConfig: this.editorConfig
    });
  }

}

function prepareMiddleware() {
  const tinymcePath = path.dirname(require.resolve('tinymce'));
  const app = express();
  app.use('/tinymce-assets', express.static(tinymcePath));
  return app;
}

let Wysiwyg = {
  type: 'Wysiwyg',
  implementation: WysiwygImplementation,
  views: {
    Controller: fields.Text.views.Controller,
    Field: _join(__dirname, "11z4har"),
    Filter: fields.Text.views.Filter
  },
  adapters: fields.Text.adapters,
  prepareMiddleware
};
exports.Wysiwyg = Wysiwyg;
